diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 PAL Robotics S.L.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * Author: Víctor López
+ */
+
+#ifndef ROLLING_MEAN_ACCUMLATOR_HPP
+#define ROLLING_MEAN_ACCUMLATOR_HPP
+
+#include <vector>
+#include <cassert>
+
+namespace diff_drive_controller
+{
+
+/**
+ * \brief Simplification of boost::accumulators::accumulator_set<double,
+ *  bacc::stats<bacc::tag::rolling_mean>> to avoid dragging boost dependencies
+ *
+ * Computes the mean of the last accumulated elements
+ */
+template<typename T>
+class RollingMeanAccumulator
+{
+public:
+  explicit RollingMeanAccumulator(size_t rolling_window_size)
+  : rolling_window_size_(rolling_window_size)
+  {
+    buffer_.reserve(rolling_window_size);
```

Comment:
would it make sense to use `resize` or the initializer constructor to default initialize the values to 0?
That would remove the need for branching later on as the buffer is always completely filled.

Reply:
Yeah, that's a great idea.

I do need to keep one branching in there to prevent the counter from overflowing and to know the actual window size for the rolling mean computation.

The size_t overflow is a far feched possibility, but I'd rather take it into account. Even though if this accumulator is going to only be used here for odometry, the chance of it overflowing in our lifetime is essentially 0.

Reply:
I made some changes that avoid using branching as far as I could see on godbolt.org with optimizations. I'd like your feedback.